File: src/GameBot.Service/Models/Sessions.cs
// Suppress CA1515: DTOs are public for controller model binding
#pragma warning disable CA1515
using System.Collections.ObjectModel;
using GameBot.Domain.Sessions;

namespace GameBot.Service.Models;

public sealed class CreateSessionRequest {
  public string? GameId { get; set; }
  public string? GamePath { get; set; }
  public string? AdbSerial { get; set; }
}

public sealed class CreateSessionResponse {
  public required string Id { get; init; }
  public required string SessionId { get; init; }
  public required string Status { get; init; }
  public required string GameId { get; init; }
}

public sealed class InputActionsRequest {
  public required Collection<InputActionDto> Actions { get; init; }
}

public sealed class InputActionDto {
  public required string Type { get; init; }
  public Dictionary<string, object> Args { get; init; } = new();
  public int? DelayMs { get; init; }
  public int? DurationMs { get; init; }
}

public sealed class RunningSessionsResponse {
  public required IReadOnlyCollection<RunningSessionDto> Sessions { get; init; }
}

public sealed class RunningSessionDto {
  public required string SessionId { get; init; }
  public required string GameId { get; init; }
  public required string EmulatorId { get; init; }
  public DateTime StartedAtUtc { get; init; }
  public DateTime LastHeartbeatUtc { get; init; }
  public RunningSessionStatus Status { get; init; }
}

public sealed class StartSessionRequest {
  public required string GameId { get; init; }
  public required string EmulatorId { get; init; }
  public Dictionary<string, object>? Options { get; init; }
}

public sealed class StartSessionResponse {
  public required string SessionId { get; init; }
  public required IReadOnlyCollection<RunningSessionDto> RunningSessions { get; init; }
}

public sealed class StopSessionRequest {
  public required string SessionId { get; init; }
}
#pragma warning restore CA1515

File: src/web-ui/src/api/sessions.ts
import axios from "axios";

export interface RunningSessionDto {
  sessionId: string;
  gameId: string;
  emulatorId: string;
  startedAtUtc: string;
  lastHeartbeatUtc: string;
  status: "running" | "stopping";
}

export interface RunningSessionsResponse {
  sessions: RunningSessionDto[];
}

export interface StartSessionRequest {
  gameId: string;
  emulatorId: string;
  options?: Record<string, unknown>;
}

export interface StartSessionResponse {
  sessionId: string;
  runningSessions: RunningSessionDto[];
}

export interface StopSessionRequest {
  sessionId: string;
}

export async function getRunningSessions() {
  const resp = await axios.get<RunningSessionsResponse>("/api/sessions/running");
  return resp.data;
}

export async function startSession(payload: StartSessionRequest) {
  const resp = await axios.post<StartSessionResponse>("/api/sessions/start", payload);
  return resp.data;
}

export async function stopSession(payload: StopSessionRequest) {
  const resp = await axios.post<{ stopped: boolean }>("/api/sessions/stop", payload);
  return resp.data;
}
